### ISSUE TITLE
Able to receive messages from contact securely, but cannot send securely.

### ISSUE THREAD

**Issue by joshproehl** [WhisperSystems/Signal-Android#5444]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
A contact which I have previously been communicating with via the secure data channel has suddenly only become able to send via SMS, however messages (and voice calls) from this contact continue to be received via the data channel. It appears that my device has decided that the remote user is no longer using Signal.
Unfortunately I was unable to capture a debug log when the changeover occurred. (27 hours ago) Looking back through the conversation it appears that a message failed to deliver, and all further messages fell back to the SMS transport. 

I have had the remote user restart the secure session, which has no impact.

The only debug step I have not taken is to delete the conversation and attempt to restart it. I would prefer not to do this and lose the conversation history if possible.

### Steps to reproduce
- Attempt to send user a message
**Actual result:** Message delivered via SMS (If message sent from desktop app a "Message not sent"  error appears.)
**Expected result:**  Message delivered via data channel. 

- Attempt to reset secure session using menu in conversation.
**Actual result:** "Invite" menu option is available.
**Expected result:**  "Reset secure session" menu option is available.

- Attempt to call user
**Actual result:** Call is made using standard dialer app
**Expected result:**  Secure call is initiated.

- Attempt to reset secure session via desktop app
**Actual result:** "Message not sent" alert appears. Re-sending causes same error.
**Expected result:**  Secure session reset successfully. 

### Device info
**Device**  : motorola XT1575 (clark_retus)
**Android version** : 6.0 (3, MPHS24.49-18-3)
***Signal version:** 3.15.2

### Link to debug log
Log taken just after attempting to send a message to the user:
https://gist.github.com/anonymous/ae2f72e176088995015b1de6db89f988

**Comment by FeuRenard** [comment]
Did you already have a look at this?
http://support.whispersystems.org/hc/en-us/articles/215089628-Why-can-I-only-privately-message-one-way-

**Comment by joshproehl** [comment]
I did take a look at that, which I forgot to mention in my bug report. However, after attempting some of the outlined procedures a second time I was able to fix the problem.

For the record:
The first time I looked at it I reset my cache, nothing happened, and then I had the remote user re-register. The problem continued at this point.
I then had the user de-register, then cleared the cache and address book on my end, and then had the user re-register. At this point it began working again. I am unable to determine if it was a specific order-of-operations that resolved it, or simply having the user re-register a second time. 

In any case, this issue is now resolved, thank you!
